src/schoology-extractor/schoology_extractor/helpers/sync.py

```diff
@@ -204,3 +214,63 @@ def sync_resource(
     _remove_duplicates(resource_name, db_engine)
 
     return DataFrame(data)
+
+
+def _create_usage_table_if_it_does_not_exist(
+        db_engine: sqlalchemy.engine.base.Engine
+        ):
+    if _table_exist(USAGE_TABLE_NAME, db_engine) is False:
+        db_engine.execute(f"""CREATE TABLE {USAGE_TABLE_NAME} (
+            FILE_NAME TEXT PRIMARY KEY
+        );""")
+
+
+def usage_file_is_processed(
+        file_name: str,
+        db_engine: sqlalchemy.engine.base.Engine
+        ) -> bool:
+    """
+    Verifies if the current file has been previously processed
+
+    Parameters
+    ----------
+    file_name: str
+        Name of the file
+    db_engine: sqlalchemy.engine.base.Engine
+
+    Returns
+    -------
+    bool
+        True if the file has been processed, False if it has not.
+    """
+
+    _create_usage_table_if_it_does_not_exist(db_engine)
+
+    query = f"SELECT * FROM {USAGE_TABLE_NAME} where FILE_NAME='{file_name}' limit 1;"
```

Comment:
I just did a little experiment and found that the following syntax will work. This is a random query that I wrote just to give the idea. The nice thing is you always get a single row back, with value of 1 or 1. You could adapt this to simplify the code a little bit. But I would save this for tomorrow, after everything else already working well. Get the whole app working now and then optimize tomorrow.
```
select exists(select 1 from assignment where assignment.SourceId = 1)
```